File: src/main/java/net/benwoodworth/fastcraft/core/api/event/events/FcCraftEvent.java
package net.benwoodworth.fastcraft.core.api.event.events;

import net.benwoodworth.fastcraft.core.api.event.FcEvent;
import net.benwoodworth.fastcraft.core.api.inventory.FcItem;
import net.benwoodworth.fastcraft.core.api.inventory.ItemGrid;

import java.util.List;

/**
 * Event for crafting items.
 */
public interface FcCraftEvent<TBase, TFcItem extends FcItem<?>> extends FcEvent<TBase> {

    /**
     * The inventory that was used to craft an item.
     */
    ItemGrid<TFcItem> getMatrix();

    /**
     * The crafting item results.
     */
    List<TFcItem> getResults();
}

File: src/main/kotlin/net/benwoodworth/fastcraft/core/api/event/events/FcCraftEvent.kt
package net.benwoodworth.fastcraft.core.api.event.events

import net.benwoodworth.fastcraft.core.api.event.FcEvent
import net.benwoodworth.fastcraft.core.api.inventory.FcItem
import net.benwoodworth.fastcraft.core.api.inventory.ItemGrid

/**
 * Event for crafting items.
 *
 * @param TBase The base event from the server platform.
 * @param TItem The type of item.
 */
interface FcCraftEvent<out TBase, out TItem : FcItem<*>> : FcEvent<TBase> {

    /**
     * The inventory that was used to craft an item.
     */
    val matrix: ItemGrid<TItem>

    /**
     * The crafting item results.
     */
    val results: List<TItem>
}
